gTea/app/src/main/java/com/nechaieva/gtea/MenuFragment.java

```diff
@@ -54,13 +57,45 @@ public void onClick(View view) {
                 if (chosen_item == RecyclerView.NO_POSITION) {
                     Toast toast = Toast.makeText(getContext(), "No item chosen", Toast.LENGTH_SHORT);
                     toast.show();
-                    return;
                 }
-                Bundle bundle = new Bundle();
-                bundle.putString("order", data[chosen_item]);
-                NavHostFragment.findNavController(MenuFragment.this)
-                        .navigate(R.id.action_makeOrder, bundle);
+                else {
+                    navigateToOrder(data[chosen_item]);
```

Comment:
Did you consider to verify the size of the array first?

Reply:
Hm. I don't see how it would get out of range provided that the data[] array doesn't get changed, but I added verification anyway.